### ISSUE TITLE
Keyboard transition shows Scaffold background color

### ISSUE THREAD

**Issue by Andrflor** [flutter/flutter#97054]
## Code to Reproduce

```dart
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.red,
      body: SingleChildScrollView(
        child: Container(
            color: Colors.white,
            width: MediaQuery.of(context).size.width,
            height: MediaQuery.of(context).size.height,
            child: Column(
              children: [
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
                TextFormField(),
              ],
            )),
      ),
    );
  }
```

## Step to reproduce
Click any field that will be underneath the keyboard and you should see the background color appearing.
You will see it even better when the keyboard closes.

## Demo
Like here you can see that the scaffold color is appearing instead of being transparent
<img src="https://s10.gifyu.com/images/outputd65b8fe78f9f61dc.gif" />

**Comment by maheshmnj** [comment]
Hi @Andrflor, Thanks for filing the issue. I am able to reproduce the same behavior on stable and the master channel.

**Comment by HansMuller** [comment]
This may an example of aknown problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
https://github.com/flutter/flutter/issues/14288#issuecomment-361375791
https://github.com/flutter/flutter/issues/17084#issuecomment-385718108
https://github.com/flutter/flutter/issues/15035#issuecomment-370028740

**Comment by maheshmnj** [comment]
Closing as duplicate of #14288
